client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js

```diff
@@ -152,10 +152,19 @@ const getPackagesErrors = ( values ) => mapValues( values, ( pckg ) => {
 		errors.box_id = translate( 'Please select a package' );
 	}
 
-	if ( ! pckg.weight || 'number' !== typeof pckg.weight || 0 > pckg.weight ) {
+	if ( isNaN( pckg.weight ) || ! isNumber( pckg.weight ) || 0 >= pckg.weight ) {
```

Comment:
I think that isNaN and ! isNumber can be merged to isFinite as described in isNumber [documentation].(https://lodash.com/docs/4.17.4#isNumber)